### DISCUSSION TITLE
More details lineage reconstruction algorithm

### DISCUSSION THREAD

**Discussion by NoELJ22** [milaboratory/mixcr#1497] [Q&A] [answered]
Hello, 
I'd like to know more about the lineage reconstruction algorithm . 
In the documentation for the `findShmTrees` command it's stated that grouped clones don't have "too distant NDN regions". I would like to know how is it determined that the regions are not too distant? is the distance between the regions calculated and then there is a fixed cut off?

Thank you for help

**Comment by mizraelson** [comment, marked as answer]
Briefly:
1) In the first step, all NDN regions are aligned with each other, incurring a certain penalty for each mismatch. These penalty scores are then normalized by the length of the region. If this normalized metric is lower than the threshold set by `-Osteps[0].algorithm.maxNDNDistanceForClustering=1.0`, the clones will be clustered.

2) In the second step, MiXCR attempts to add additional members to the tree without disrupting its structure. The parameter `-Ostep[1].maxNDNDistance=1.5` controls the maximum distance allowed between the NDN-region of the added clonotypes and the closest existing tree member.

3) In the third step, MiXCR tries to merge entire trees that have similar most recent common ancestors. The parameter `-Ostep[2].maxNDNDistanceBetweenRoots=0.5` determines the required level of similarity in the NDN region for this merging to occur.

Generally speaking, increasing the values mentioned above will result in more aggressive clustering.

These parameters are also described [here](https://mixcr.com/mixcr/reference/mixcr-findShmTrees/#lineage-reconstruction-algorithm-parameters).

**Comment by NoELJ22** [comment]
Thank you for your answer.

Do you think it might be relevant to test less stringent parameters to try and group together clones that may have accumulated a lot of mutations?

In the context of our study, we are tracking clones at D0 and D7 after stimulation, in vitro.
With lineage analysis, we find some clusters with only clones at D7, without clones from D0. So we wondered whether this could be due to a greater accumulation of mutations.

**Reply by mizraelson** [reply]
You can certainly play with it. At the same time it is common that you do not see clones from D7 in D0 because these lineages might have arise from single cells that you didn't capture in D0.

**Reply by NoELJ22** [reply]
Ok, thank you for these informations.
